### ISSUE TITLE
Enable RazorCompileOnBuild=true by default

### ISSUE THREAD

**Issue by rynowak** [aspnet/Razor#1986]
Due to how our factoring is being done in 2.1-preview1 - the necessary properties to activate Razor  compilation (using RazorSDK) during publish are part of the Web SDK. This means that a class library project won't do precompilation during publishing by default.

The workaround is to define a property that the WebSDK sets.
```
  <PropertyGroup>
    <TargetFramework>netcoreapp2.0</TargetFramework>
    <ResolvedRazorCompileToolset>RazorSDK</ResolvedRazorCompileToolset>
  </PropertyGroup>
```

We plan to take ownership of these definitions from the WebSDK in 2.1-preview2 so this should just get solved at that point.

**Comment by rynowak** [comment]
@DamianEdwards - do you have concerns about this for preview1?

**Comment by DamianEdwards** [comment]
No this is fine, we'll be sure to document the requirements for preview1

**Comment by rynowak** [comment]
Stashing this in preview2 then - this tracks removing a workaround.

**Comment by rynowak** [comment]
@pranavkm assigning you. We plan to leave the current behaviour for preview1, but this can be fixed now in preview2 since we control the evaluation order.

**Comment by pranavkm** [comment]
The last bit to enable here is to turn on `RazorCompileOnBuild = true`. This is blocked pending further details from inner-loop perf results.

**Comment by rynowak** [comment]
Cool beans! Changing the title on this spicy boy!

**Comment by rynowak** [comment]
Also, we have the perf numbers and everything looks good from that point of view. We'll do an experience review next week and then we'll be ready to go.

**Comment by mkArtakMSFT** [comment]
@pranavkm, is this unblocked already?

**Comment by pranavkm** [comment]
The experience review is unblocked and @ajaybhargavb's per PR also went in which was another blocked. I'd think this is. @rynowak any concerns?
